cluster-autoscaler/core/utils/utils.go

```diff
@@ -139,6 +140,40 @@ func sanitizeTemplateNode(node *apiv1.Node, nodeGroup string, ignoredTaints tain
 	return newNode, nil
 }
 
+func PodSpecSemanticallyEqual(p1 apiv1.PodSpec, p2 apiv1.PodSpec) bool {
+	p1Spec := sanitizedProjectedVolumesAndMounts(p1)
+	p2Spec := sanitizedProjectedVolumesAndMounts(p2)
+	return apiequality.Semantic.DeepEqual(p1Spec, p2Spec)
+}
+
+// sanitizedProjectedVolumesAndMounts returns a pod spec with projected volumes
+// and their mounts removed
+// Due to the generated suffixes, a strict DeepEquals check will fail and generate
+// an equivalence group per pod which is undesirable.
+// Projected volumes do not impact scheduling so we should ignore them
+func sanitizedProjectedVolumesAndMounts(podSpec apiv1.PodSpec) apiv1.PodSpec {
+	var projectedVolumeNames []string
+	for i, v := range podSpec.Volumes {
+		if v.Projected != nil {
+			projectedVolumeNames = append(projectedVolumeNames, v.Name)
+			podSpec.Volumes[i] = apiv1.Volume{}
```

Comment:
preference for niling them out vs keeping a slice with the "good" volumes we want to keep and then set that as the volumes slice?

Just went with niling out for the sake of simplicity.